### ISSUE TITLE
Device session is not exposed after system sleep.

### ISSUE THREAD

**Issue by GiviMAD** [jellyfin/jellyfin-androidtv#1336]
### Describe the bug

Device is not exposed as controllable in the web interface after a (device) system sleep.
I have discovered this issue in a Xiaomi TV as it sends the system to sleep when you turn off the tv. I know it's also happening on the Xiaomi Mi Box, and I can replicate the issue o my Fire TV if a press go to sleep on the menu, as I far I saw my fire tv never sleeps as less I press this option.
No idea why this is happening, maybe the device is trying to update its old session instead of creating a new one.
I can workaround this force closing the app from the apps menu or using the adb.
I have tested this on the current version 0.12.3 and compiling version from the main branch (yesterday), same happens.
Hope some of this helps you to find the issue.
Regards! 

### Application version

0.12.3

### Device information

MiTV-MSSP2 (sideload), Mi box (from store), Amazon Fire TV Stick 4K Max (from store)

### Android version

9

### Jellyfin server version

10.7.7

**Comment by nielsvanvelzen** [comment]
When an Android device (especially with low processors like televisions / tv sticks) is turned "off" (put to sleep) they automatically suspend apps, a lot even disable the network connection. When this happens the device won't be active in the Jellyfin server until the app is opened again. Not much we can do about that.

**Comment by GiviMAD** [comment]
Yes, I understand that's normal. The problem is that when I turn on the tv again I can continue using the app but the session is not exposed anymore unless I force close the app and launch it again.

**Comment by nielsvanvelzen** [comment]
I think I'm able to reproduce this issue now. Does it work when you go to the "change user" screen and select the same user as before?

**Comment by GiviMAD** [comment]
Yes, if I go to "change user" and change to the same user (my only user at the moment), the remote control starts working again.

**Comment by nielsvanvelzen** [comment]
Can you test if this branch fixes it for you:
https://github.com/nielsvanvelzen/jellyfin-androidtv/tree/restore-connection-on-start

It reverts #1074 partially so going back from the home screen doesn't behave properly.

**Comment by GiviMAD** [comment]
I have testes it and something has changed, if I go to the launcher and back to the app the problem it solved, no need to force close. 
But still if the app is opened when I turn off the tv I can turn it on again and use the app and the remote control doesn't work.

I was taking a look to your branch and I think that if we change this code app/src/main/java/org/jellyfin/androidtv/JellyfinApplication.kt
```
	/**
	 * Called by the Process Lifecycle when the app is activated in the foreground (activity opened).
	 */
	@OnLifecycleEvent(Lifecycle.Event.ON_START)
	fun onActivate() {
		Timber.i("Process Lifecycle started")

		get<SessionRepository>().restoreDefaultSession()
	}
```
to also listen for the ON_RESUME event the problem will be solved, because I think the onStart event is not triggered after a device sleep. Do you think that's makes sense?

**Comment by mueslimak3r** [comment]
> to also listen for the ON_RESUME event the problem will be solved, because I think the onStart event is not triggered after a device sleep in this situation (device coming back from sleep while the app is opened). Do you think this makes sense?

In that case you wouldn't do ON_RESUME in addition to ON_START. You would only do resume. This is because the lifecycle goes from start -> resume
